### ISSUE TITLE
Wrong file chosen when comparing case sensitivity for import/no-unresolved

### ISSUE THREAD

**Issue by jimmymeraki** [import-js/eslint-plugin-import#2335]
I came across a situation where a file named "OWNERS" and a file named "Owners.jsx" in the same directory caused the "import/no-unresolved" rule to report a case mismatch when another module imported "Owners". Removing or renaming the file named "OWNERS" prevents the error. This behavior can be reproduced with the following minimal example:

## Setup

Given the following files in a directory:

package.json:

```json
{
  "devDependencies": {
    "eslint": "^8.5.0",
    "eslint-plugin-import": "^2.25.3"
  }
}
```

.eslintrc.yml:

```yml
parserOptions:
  ecmaVersion: 2015
  sourceType: module
plugins:
  - import
rules:
  import/no-unresolved: 2
```

src/Foo.js:

```js
export default function Foo() {
}
```

src/Bar.js:

```js
import Foo from "./Foo";

export default function Bar() {
  return Foo();
}
```

src/FOO:

(contents are irrelevant)

## Action

Run in a shell:

```
yarn install
yarn run eslint src
```

## Expected behavior

No errors should be reported.

## Actual behavior

An error is reported:

```
/path/to/example/src/Bar.js
  1:17  error  Casing of ./Foo does not match the underlying filesystem  import/no-unresolved

✖ 1 problem (1 error, 0 warnings)
```

## Additional info

* As mentioned at the top of the issue, if the file at `src/FOO` is removed from the example setup, running `yarn run eslint` reports no errors, as expected.
* The problem is reproducible on macOS 10.15.7 on a case-insensitive APFS volume, but not on Linux on a case-sensitive volume. This suggests file system case sensitivity is a necessary condition for the problem.

**Comment by tyuqing** [comment]
me too

**Comment by ljharb** [comment]
Actually I think this is a valid warning.

Essentially, for a specifier `Foo`, when a file `Foo` exists with matching casing, `Foo.js` will be ignored. Having a file named "FOO", then, means that on a case-insensitive filesystem, your import will grab FOO instead of Foo.js, and the linter warning is thus correct.

I'll add this test case - warning, as expected - to close the issue.

**Comment by ljharb** [comment]
oops, looks like indeed the test does fail to warn on a case-sensitive volume, so i'll use that to fix it.

**Comment by ljharb** [comment]
@jimmymeraki right - on your system it's correct, but on a case INSENSITIVE system, i think it would resolve to OWNERS.

**Comment by ljharb** [comment]
Interesting, hmm 👀
